src/app/api/documents/route.ts

```
@@ -0,0 +1,207 @@
+    const page = parseInt(searchParams.get('page') || '1', 10);
+    const pageSize = parseInt(searchParams.get('page_size') || '10', 10);
+    const maxPageSize = 100;
+
+    if (isNaN(page) || page < 1) {
+      return NextResponse.json(
+        { error: "Invalid 'page' parameter. Must be a positive integer." },
+        { status: 400 }
+      );
+    }
+
+    if (isNaN(pageSize) || pageSize < 1 || pageSize > maxPageSize) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)